sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -250,29 +250,12 @@ private RxDocumentClientImpl(URI serviceEndpoint,
     }
 
     private void initializeGatewayConfigurationReader() {
-        String resourceToken;
-        if(this.tokenResolver != null) {
-            resourceToken = this.tokenResolver.getAuthorizationToken(RequestVerb.GET, "", CosmosResourceType.System, null);
-        } else if(!this.hasAuthKeyResourceToken && this.authorizationTokenProvider == null) {
-            resourceToken = this.firstResourceTokenFromPermissionFeed;
-        } else {
-            assert  this.masterKeyOrResourceToken != null || this.cosmosKeyCredential != null;
-            resourceToken = this.masterKeyOrResourceToken;
-        }
-
-        this.gatewayConfigurationReader = new GatewayServiceConfigurationReader(this.serviceEndpoint,
-                this.hasAuthKeyResourceToken,
-                resourceToken,
-                this.connectionPolicy,
-                this.authorizationTokenProvider,
-                this.reactorHttpClient);
-
-        DatabaseAccount databaseAccount = this.gatewayConfigurationReader.initializeReaderAsync().block();
+        this.gatewayConfigurationReader = new GatewayServiceConfigurationReader(this.serviceEndpoint, this.globalEndpointManager);
+        DatabaseAccount databaseAccount = this.globalEndpointManager.getLatestDatabaseAccount();
```

Comment:
where are we doing the initialization blocking call now?

If the global endpoint manager is not fully initialized with first databaseAccount fetch could this return null?

Reply:
GatewayServiceConfigurationReader constructor is doing blocking call on async cache , so for first time if cache in null it will hit the BE

Reply:
could you please add an `assert(databaseAccount != null)` here. for sanity check and readibility.

Reply:
done